.github/workflows/ci-docs.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Docs
+
+on:
+  pull_request:
+  push:
+
+env:
+  STRUCTOR_VERSION: v1.7.1
+  BUILD_MAVEN_OPTS: "-Dmaven.javadoc.skip=true -DskipTests=true --batch-mode --also-make"
+  VERIFY_MAVEN_OPTS: "-Pcoverage"
+
+jobs:
+  build-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install --projects janusgraph-all ${{ env.BUILD_MAVEN_OPTS }}
+
+  build-doc:
+    runs-on: ubuntu-20.04
+    needs: build-all
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install -pl janusgraph-doc ${{ env.BUILD_MAVEN_OPTS }}
+      - run: git diff  --exit-code docs/basics/janusgraph-cfg.md
+      - run: docker build -t doc-site:mkdocs -f docs/docs.Dockerfile ./docs
+      - run: docker run --rm -v $PWD:/mkdocs doc-site:mkdocs mkdocs build
+      - uses: actions/upload-artifact@v2
+        if: github.ref != 'ref/head/master'
+        with:
+          name: distribution-builds
+          path: site
+
+  deploy-doc:
+    runs-on: ubuntu-20.04
+    if: github.ref == 'ref/head/master'
+    needs: build-doc
+    steps: 
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: mkdir ./bin
+      - run: curl -sfL https://raw.githubusercontent.com/traefik/structor/master/godownloader.sh | bash -s -- -b ./bin ${STRUCTOR_VERSION}
+      # TODO: change --dockerfile-url=".../v0.5/docs.Dockerfile" to --dockerfile-url=".../master/docs/docs.Dockerfile"
```

Comment:
When should we change this url? Why not to change it right now? This PR is targeting `master` branch

Reply:
The file doesn't exists right now in master. It will be create with this PR.

Reply:
I didn't research structor flow. I see that the file exists: https://raw.githubusercontent.com/janusgraph/janusgraph/master/docs.Dockerfile

Maybe I'm just missing something

Reply:
I think I understand. Most likely we will need to change the URL with a separate PR. I'm good with it